builder_test.go

```diff
@@ -310,36 +310,286 @@ func testBuilder(t *testing.T, when spec.G, it spec.S) {
 							t.Fatalf("Unexpected error:\n%s\n", err)
 						}
 						if s := cmp.Diff(metadata.Processes, []launch.Process{
+							{
+								Type:        "some-type",
+								Command:     "some-command",
+								Args:        []string{"some-arg"},
+								Direct:      true,
+								BuildpackID: "A",
+							},
+							{
+								Type:        "some-other-type",
+								Command:     "some-other-command",
+								Args:        []string{"some-other-arg"},
+								Direct:      true,
+								BuildpackID: "B",
+							},
 							{
 								Type:        "override-type",
 								Command:     "bpB-command",
 								Args:        []string{"bpB-arg"},
 								Direct:      false,
 								BuildpackID: "B",
 							},
+						}); s != "" {
+							t.Fatalf("Unexpected:\n%s\n", s)
+						}
+					})
+
+					it("should warn when overriding default process type, with non-default process type", func() {
+						bpA := testmock.NewMockBuildpack(mockCtrl, builder.Group.Group[0].API)
+						buildpackStore.EXPECT().Lookup("A", "v1").Return(bpA, nil)
+						bpA.EXPECT().Build(gomock.Any(), config).Return(lifecycle.BuildResult{
+							Processes: []launch.Process{
+								{
+									Type:        "override-type",
```

Comment:
What about `override-with-default` and `override-with-not-default` for these process names? That might make the test a bit easier to read and understand.

Reply:
Please see https://github.com/buildpacks/lifecycle/pull/520/commits/63f4e9a470ff92f133caa5ffff56d047d6fa3295.